rust/private/rustc.bzl

```diff
@@ -617,7 +617,7 @@ def construct_arguments(
         rustc_flags.add(linker_script.path, format = "--codegen=link-arg=-T%s")
 
     # Gets the paths to the folders containing the standard library (or libcore)
-    rust_std_paths = depset([file.dirname for file in toolchain.rust_std.files.to_list()]).to_list()
+    rust_std_paths = depset([file.dirname for file in toolchain.rust_std.to_list()]).to_list()
```

Comment:
Since you're thinking about the toolchain these days, WDYT about preparing this depset when creating the toolchain (in other words adding `rustc_std_paths` field to the `RustToolchainInfo`), and not when creating a rustc action? Seems wasteful.

Reply:
Yeah, good call. Also, is there `RustToolchainInfo`? Is that is that what's created by `platform_common.ToolchainInfo` or is this a suggestion for a new provider?

Reply:
Wait, can [dirname](https://docs.bazel.build/versions/main/skylark/lib/File.html#dirname) be calculated like this in the toolchain? 

> The name of the directory containing this file. It's taken from [path](https://docs.bazel.build/versions/main/skylark/lib/File.html#path) and is always relative to the execution directory. 

Couldn't that potentially be different based on the target? Even if the toolchain was the same? Maybe that would imply a loosely constrained toolchain? Does any of that make any sense? 😅

Reply:
* Whoops, no there is no `RustToolchainInfo`, it's just `ToolchainInfo`. Bazel toolchains and platforms don't have the ability to specify which provider type is returned from `ctx.toolchains`, it's always `ToolchainInfo`, with custom fields.
* If you provide a `File` from a target, its `dirname` will be constant, no matter which target depends on it (no matter what target is "asking"). So yup, you can calculate that dirname in the toolchain. I'm not really following the issue you're  anticipating to the detail, but I'd say we're safe here anyway.

Reply:
Continued in https://github.com/bazelbuild/rules_rust/pull/1115